### DISCUSSION TITLE
Online waiting! about to started clangd use compile_commands.json from LSP initialize request

### DISCUSSION THREAD

**Discussion by Yurii-huang** [clangd/clangd#1152] [Q&A] [answered]
The source protocol (LSP) initialize request does not seem to specify the backend compilation data configuration file (compile_commands. json). This json file is unique to Clangd, right?

If I want to compile the specified data configuration file, can I extend the initialize request intelligently?

**Comment by HighCommander4** [comment, marked as answer]
Clangd supports an extension to specify the compilation database directory in the initialize request, see https://clangd.llvm.org/extensions#compilation-commands

**Comment by Yurii-huang** [comment]
Thank you very much for your reply.
